app/controllers/gazetteers_controller.rb

```diff
@@ -0,0 +1,143 @@
+class GazetteersController < ApplicationController
+  include DataControllerConfiguration::ProjectDataControllerConfiguration
+  before_action :set_gazetteer, only: %i[ show edit update destroy ]
+
+  # GET /gazetteers
+  # GET /gazetteers.json
+  def index
+    respond_to do |format|
+      format.html do
+        @recent_objects = Gazetteer.recent_from_project_id(sessions_current_project_id).order(updated_at: :desc).limit(10)
+        render '/shared/data/all/index'
+      end
+      format.json do
+        # TODO gzs, not gas
+        @geographic_areas = ::Queries::GeographicArea::Filter.new(params).all
+          .includes(:geographic_items)
+          .page(params[:page])
+          .per(params[:per])
+          # .order('geographic_items.cached_total_area, geographic_area.name')
+      end
+    end
+  end
+
+  # GET /gazetteers/1 or /gazetteers/1.json
+  def show
+  end
+
+  # GET /gazetteers/new
+  def new
+    respond_to do |format|
+      format.html { redirect_to new_gazetteer_task_path }
+    end
+  end
+
+  # GET /gazetteers/1/edit
+  def edit
+    respond_to do |format|
+      format.html { redirect_to new_gazetteer_task_path gazetteer_id: @gazetteer.id }
+    end
+  end
+
+  # GET /gazetteers/list
+  def list
+    @gazetteers = Gazetteer
+      .with_project_id(sessions_current_project_id)
+      .page(params[:page]).per(params[[:per]])
+  end
+
+  # POST /gazetteers.json
+  def create
+    @gazetteer = Gazetteer.new(gazetteer_params)
+
+    begin
```

Comment:
Please move this into Gazetteer.  See older `shape` variable for processing.

Reply:
I think I got it, but please let me know if it still needs some work :)